script.js

```diff
@@ -1,12 +1,216 @@
+  
+
+  getAllShows().sort((a, b) => (a.name > b.name) ? 1 : ((b.name > a.name) ? -1 : 0)).forEach((item) => {
+
+    let opt = item.name;;
+   
+    let el = document.createElement("option");
+    el.textContent = opt;
+    el.value = opt;
+    selectShows.appendChild(el);
+
+  });
 }
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+
+//sort the shows
+
+
+
+///Now I have all the shows so need to  get access to  one show which user selected in select box.
+
+function selectShowsToGetEpisodes() {
+  document.querySelector("#select-shows").addEventListener('change', (e) => {
+    const selectString = e.target.value;
+    console.log(selectString);
+    let selectedShow = getAllShows().filter((item) => {
+      
+      return (
+        item.name === selectString);
+    });
+    
+    let idOfShow = selectedShow.map(data => data.id);
+  
+    //Now I have an access to a particular shows that user serlect from the select box so I need 
+    //to go and fetch episodes of that selected show.
+
+    //fetch an api based on selcted show
+    fetch(`https://api.tvmaze.com/shows/${idOfShow}/episodes`)
+      .then((response) => response.json())
+      .then((episodesOfShow) => {
+        displayEpisodes(episodesOfShow);
+        searchEpisodes(episodesOfShow);
+        populateEpisodes(episodesOfShow);
+        selectEpisodesToDisplay(episodesOfShow);
+        goBackToAllEpisodesFunction(episodesOfShow);
+      });
+  });
+  
+
+
+}
+
+//implement the search functionality for the selected shows episodes
+function searchEpisodes(episodes) {
+  let totalEpisodes = episodes.length;
+  let total = document.querySelector(".total-1")
+  total.textContent = totalEpisodes;
+  searchBar.addEventListener('keyup', (e) => {
+
+    const searchString = e.target.value.toLowerCase();
+    const filteredMovies = episodes.filter((item) => {
+      return (
+        item.name.toLowerCase().includes(searchString) ||
+        item.summary.toLowerCase().includes(searchString)
+      );
+    });
+    displayEpisodes(filteredMovies);
+  });
+}
+
+//populate selectbox with corrosponding epsodes
+
+function populateEpisodes(allEpisodes) {
+  selectEpisodes.length = 0;
+allEpisodes.forEach((item) => {
+    
+    let opt = `S${item.season.toString().padStart(2,0)}E${item.number.toString().padStart(2,0)}-${item.name}`;
+   
+  
+  
+  let element = document.createElement('option')
+    element.textContent = opt;
+    element.value = opt;
+    selectEpisodes.add(element);
+
+  });
+ 
+ 
 }
 
-window.onload = setup;
+//Now I have all the episodes of corrosponding show ,I need to find a 
+//way to display them.
+function displayEpisodes(allShows) {
+  let EpisodeBeingDisplay = allShows.length;
+  let total = document.querySelector(".total-2")
+  total.textContent = EpisodeBeingDisplay;
+  
+  let displayEpisode = allShows.map((item) => {
```

Comment:
I'd consider thinking about creating a function like `create SingleEpisode` that returns the html you've created here